### ISSUE TITLE
Change gauge command structure to have subcommands with flags

### ISSUE THREAD

**Issue by kashishm** [getgauge/gauge#676]
## Old Usage

```
Gauge version 0.8.4
Copyright 2017 ThoughtWorks, Inc.

Usage:
	gauge specs/
	gauge specs/spec_name.spec

Options:
  --add-plugin=""                  Adds the specified non-language plugin to the current project
  --api-port=""                    Specifies the api port to be used. Eg: gauge --daemonize --api-port 7777
  --check-updates=false            Checks for Gauge and plugins updates. Eg: gauge --check-updates
  --daemonize=false                Run as a daemon
  --dir="."                        Set the working directory for the current command, accepts a path relative to current directory.
  --docs=""                        Generate documenation using specified plugin. Eg: gauge --docs <plugin name> specs/
  --env="default"                  Specifies the environment. If not specified, default will be used
  --failed=false                   Run only the scenarios failed in previous run. Eg: gauge --failed
  --file, -f=""                    Installs the plugin from zip file. This is used with --install. Eg: gauge --install java -f ZIP_FILE
  --format=""                      Formats the specified spec files
  -g, --group=-1                   Specify which group of specification to execute based on -n flag
  --init=""                        Initializes project structure in the current directory. Eg: gauge --init java
  --install=""                     Downloads and installs a plugin. Eg: gauge --install java
  --install-all=false              Installs all the plugins specified in project manifest, if not installed. Eg: gauge --install-all
  --list-templates=false           Lists all the Gauge templates available. Eg: gauge --list-templates
  --log-level=""                   Set level of logging to debug, info, warning, error or critical
  --machine-readable=false         Used with `--version` to produce JSON output of currently installed Gauge and plugin versions. e.g: gauge --version --machine-readable
  -n=8                             Specify number of parallel execution streams
  --order, -o=""                   Set the specs execution order. Possible options are: `random`, `sort`. Ex: gauge --order="random"
  --parallel, -p=false             Execute specs in parallel
  --plugin-args=""                 Specified additional arguments to the plugin. This is used together with --add-plugin
  --plugin-version=""              Version of plugin to be installed. This is used with --install or --uninstall flag.
  --refactor=""                    Refactor steps. Eg: gauge --refactor <old step> <new step> [[spec directories]]
  --simple-console=false           Removes colouring and simplifies from the console output
  --sort, -s=false                 Run specs in Alphabetical Order. Eg: gauge -s specs
  --strategy="lazy"                Set the parallelization strategy for execution. Possible options are: `eager`, `lazy`. Ex: gauge -p --strategy="eager"
  --table-rows=""                  Executes the specs and scenarios only for the selected rows. It can be specified by range as 2-4 or as list 2,4. Eg: gauge --table-rows "1-3" specs/hello.spec
  --tags=""                        Executes the specs and scenarios tagged with given tags. Eg: gauge --tags tag1,tag2 specs
  --uninstall=""                   Uninstalls a plugin. Eg: gauge --uninstall java
  --update=""                      Updates a plugin. Eg: gauge --update java
  --update-all=false               Updates all the installed Gauge plugins. Eg: gauge --update-all
  -v, --version, -version=false    Print the current version and exit. Eg: gauge --version
  --validate=false                 Check for validation and parse errors. Eg: gauge --validate specs
  --verbose=false                  Enable step level reporting on console, default being scenario level. Eg: gauge --verbose specs
```

## New Usage

```
Gauge is a light-weight cross-platform test automation tool with the ability to author test cases in the business language.

Complete documentation is available at https://docs.getgauge.io/.

Usage:
  gauge <command> [flags] [args]

Examples:
  gauge run specs/
  gauge run -p specs/

Available Commands:
  add            Adds the specified non-language plugin to the current project.
  daemonize      Run as a daemon.
  docs           Generate documenation using specified plugin.
  format         Formats the specified spec files.
  help           Help about any command
  init           Initializes project structure in the current directory.
  install        Downloads and installs a plugin.
  list-templates Lists all the Gauge templates available.
  refactor       Refactor steps.
  run            Run specs.
  uninstall      Uninstalls a plugin.
  update         Updates a plugin.
  validate       Check for validation and parse errors.
  version        Print Gauge and plugin versions.

Flags:
  -d, --dir string         Set the working directory for the current command, accepts a path relative to current directory. (default ".")
  -h, --help               help for gauge
  -l, --log-level string   Set level of logging to debug, info, warning, error or critical. (default "info")

Use "gauge [command] --help" for more information about a command.
```

### Subcommands

* add
   * --plugin-args
* daemon
* docs
* format
* help
* init
* install
   * -a, --all
   * -f, --file string
   * -v, --plugin-version
* list-templates
* refactor
* run
   * -e, --env
   * -f, --failed
   * -g, --group
   * -n, --n
   * -p, --parallel
   * --simple-console
   * -s, --sort
   * --strategy
   * -r, --table-rows
   * -t, --tags
   * -v, --verbose
* uninstall
   * -v, --plugin-version
* update
   * -a, --all
   * -c, --check
* validate
* version
   * -m, --machine-readable

### Global Flags
*  -d, --dir
*  -l, --log-level

> Deprecate the old usage.

**Comment by sguptatw** [comment]
gauge -v --log-level=debug is not printing the generated Id.
where as gauge version --log-level=debug is printing it.

Warning: '-v' is deprecated, it will be removed soon. See usage.
Warning: '--log-level' is deprecated, it will be removed soon. See usage.
Gauge version: 0.9.0.nightly-2017-06-28

Plugins
-------
html-report (4.0.1.nightly-2017-06-27)
java (0.6.5.nightly-2017-06-27)
js (2.0.0)
python (0.2.0)
ruby (0.4.1.nightly-2017-05-08)
xml-report (0.2.0)

**Comment by sguptatw** [comment]
Remove --order, -o="" from the gauge --help

**Comment by sguptatw** [comment]
Modify the deprecated warning message. Instead of See usage say `gauge old-help`

**Comment by sguptatw** [comment]
Run specifications with "gauge specs/" message to be changed. This occurs in the initialization of a project

**Comment by sguptatw** [comment]
gauge update -c
Is printing an error message after performing the task

**Comment by sguptatw** [comment]
gauge daemon
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/getgauge/gauge/cmd.glob..func3(0x19494a0, 0x196ef18, 0x0, 0x0)
	src/github.com/getgauge/gauge/cmd/daemon.go:28 +0x154
github.com/spf13/cobra.(*Command).execute(0x19494a0, 0x196ef18, 0x0, 0x0, 0x19494a0, 0x196ef18)
	src/github.com/spf13/cobra/command.go:651 +0x23a
github.com/spf13/cobra.(*Command).ExecuteC(0x1949280, 0x421bd9020, 0x165bb8e, 0x3)
	src/github.com/spf13/cobra/command.go:726 +0x339
github.com/getgauge/gauge/cmd.Parse(0x421b5b7b0, 0x421b5b7a0, 0x421b5b780)
	src/github.com/getgauge/gauge/cmd/cmd.go:77 +0x93
main.main()
	/Users/vagrant/go-agent-17.4.0/pipelines/Gauge-Nightly/gauge.go:84 +0x3c

**Comment by sguptatw** [comment]
**Actual behavior**
We are getting warning while `gauge init ruby`
Warning: '-v' is deprecated, it will be removed soon. See usage.
Warning: '--machine-readable' is deprecated, it will be removed soon. See usage.

**Expected behavior**
We should not get these warnings when executing it as a part of other commands

**Comment by sguptatw** [comment]
list-templates should not be a separate subcommand. It should come under init.

**Comment by sguptatw** [comment]
Grammar and spell check
* Incorrect spelling - Generate **documenation** using specified plugin
* The end of each option should not end with a full stop
* Correct to **Initialize** project structure in the current directory

**Comment by sguptatw** [comment]
`gauge uninstall` is printing the usage twice

**Comment by sguptatw** [comment]
gauge update should update everything. Remove the -all flag.

**Comment by sriv** [comment]
>  New usage should be recommended in messages
Will be addressed via #689, post release.

> Getting warnings for old usages while gauge init ruby
Will be addressed via #692, post release.
